browser/create-template/templates/nextjs-site/src/components/Searchbar.tsx

```diff
@@ -0,0 +1,70 @@
+'use client';
+
+import HStack from './Layout/HStack';
+import styles from './Searchbar.module.css';
+import FaMagnifyingGlass from './Icons/magnifying-glass-solid.svg';
+import Image from 'next/image';
+import { usePathname, useRouter, useSearchParams } from 'next/navigation';
+import { useCallback, useEffect, useRef } from 'react';
+
+const Searchbar = () => {
+  const router = useRouter();
+  const pathname = usePathname();
+  const searchParams = useSearchParams();
+
+  const inputRef = useRef<HTMLInputElement>(null);
+  const debounceTimer = useRef<NodeJS.Timeout | null>(null);
+
+  const createQueryString = useCallback(
+    (name: string, value: string) => {
+      const params = new URLSearchParams(searchParams.toString());
+      params.set(name, value);
+      return params.toString();
+    },
+    [searchParams],
+  );
+
+  const handleSearchChange = () => {
+    if (debounceTimer.current) {
+      clearTimeout(debounceTimer.current);
+    }
+
+    debounceTimer.current = setTimeout(() => {
```

Comment:
I'd suggest using an external `throttle` implementation, which is subtly faster.